File: 02_Recursion_Backtracking_&_Divide_and_Conquer/11729_Tower_of_Hanoi/Answer11729.ts
const n = +require('fs').readFileSync(0)
const resultArr = [`${2 ** n - 1}`]

const recursionFn = (n, from, to) => {
  const auxiliary = 6 - from - to
  if(n === 1) {
    resultArr.push(`${from} ${to}`)
  } else {
    recursionFn(n - 1, from, auxiliary)
    resultArr.push(`${from} ${to}`)
    recursionFn(n - 1, auxiliary, to)
  }
}
recursionFn(n, 1, 3)

console.log(resultArr.join('\n'))

File: 02_Recursion_Backtracking_&_Divide_and_Conquer/11729_Tower_of_Hanoi/Answer11729.java
import java.util.Scanner;

public class Answer11729 {
  public static void recursionFn(int n, int from, int to, StringBuilder resultStr){
    int auxiliary = 6 - from - to;
    if(n == 1){
      var s = "\n" + from + " " + to;
      resultStr.append(s);
    } else {
      recursionFn(n - 1, from, auxiliary, resultStr);
      var s = "\n" + from + " " + to;
      resultStr.append(s);
      recursionFn(n - 1, auxiliary, to, resultStr);
    }
  }

  public static void main(String[] args) {
    var n = new Scanner(System.in).nextInt();
    var resultStr = new StringBuilder();
    resultStr.append(((1 << n) - 1));

    recursionFn(n, 1, 3, resultStr);

    System.out.print(resultStr);
  }
}